packages/server/app/Controllers/Http/AuthenticationController.ts

```diff
@@ -315,4 +317,39 @@ export default class AuthenticationController {
             status: 'SUCCESS',
         };
     }
+
+    public async validatePasswordResetToken({
+        request,
+        response,
+    }: HttpContextContract): Promise<ValidatePasswordResetTokenResponseBody> {
+        const { token, email } = ValidatePasswordResetTokenRequestBody.parse(
+            request.body(),
+        );
+
+        const user = await User.findBy('email', email);
+        if (user === null) {
+            response.status(400);
+
+            return {
+                status: 'INVALID_TOKEN',
+            };
+        }
+
+        const isValidToken = await user.checkToken({
+            token,
+            tokenType: 'PASSWORD_RESET',
+        });
+        const isInvalidToken = isValidToken === false;
+        if (isInvalidToken === true) {
+            response.status(400);
+
+            return {
+                status: 'INVALID_TOKEN',
+            };
+        }
+
```

Comment:
Thoughts: user will then have to sign in using his new password ? What will do or display the client after this operation success ?

Reply:
This route will be requested at this step:

![CleanShot 2022-04-04 at 09 48 55@2x](https://user-images.githubusercontent.com/29370468/161498188-0f275b2d-1675-4e37-99d3-ff3cb1c81688.png)

It's only meant to confirm that the token is valid. After that, we redirect the user to the form where she fills her new password.
And when the user submits her new password, we send to the backend the email, the token and the new password. And we'll authenticate her automatically if the operation succeeds!